### PR TITLE
[JBPM-9323] Some form options can't be rendered in API response / MultipleSelector

### PR DESCRIPTION
**Pull request by elguardian**
JIRA: https://issues.redhat.com/browse/JBPM-9323

### CODE REVIEW

**Review by afalhambra — CHANGES_REQUESTED** [review]
Overall looks good with the exception of the `MultipleInput` which does not work in Wildfly (reason being is explained in the review). Listing below the issues found.

For `MultipleSelector`:

- There is no "Search filter" as specified in the task form on BC.
- There is no "Clear selection action" as specified in the task form on BC.
- "Elements visible on the dropdown" has no effect on the task form.
- "Elements visible on dropdown title" has no effect on the task form.
- "Required" has no effect on the task form.
- "Help message" doesn't come up.

For "MultipleInput":
- Values entered in the field are not stored, they're gone once you get back to the task even after saving or completing the task.
- "Help message" doesn't come up.
- "Page size" has no effect.
- "Required" has no effect on the task form.

**Comment by elguardian** [comment]
Hi @afalhambra  it is not required for jbpm form ui kie server to work exactly as BC

**Comment by afalhambra** [comment]
> Hi @afalhambra it is not required for jbpm form ui kie server to work exactly as BC

Ok, fine with me. However there are two main outstanding issues as mention in the review:

- `MultipleInput` doesn't get displayed when using Wildfly
- `MultipleInput` inputs are not being saved/stored once the task is either saved or complete.

Would you please have a look?

File: kie-server-parent/kie-server-services/kie-server-services-jbpm-ui/src/main/java/org/kie/server/services/jbpm/ui/form/render/AbstractFormRenderer.java [hunk 1]
```diff
@@ -16,7 +16,9 @@
 
 package org.kie.server.services.jbpm.ui.form.render;
 
+import static java.util.Arrays.asList;
 import static java.util.Collections.singletonList;
```
[the remark this answers is not in the document]
Comment (elguardian) [reply kie-server-parent/kie-server-services/kie-server-services-jbpm-ui/src/main/java/org/kie/server/services/jbpm/ui/form/render/AbstractFormRenderer.java]: done

File: kie-server-parent/kie-server-services/kie-server-services-jbpm-ui/src/main/java/org/kie/server/services/jbpm/ui/form/render/AbstractFormRenderer.java [hunk 2]
```diff
@@ -34,6 +35,7 @@
 import java.util.List;
 import java.util.Map;
 import java.util.Optional;
+import java.util.StringJoiner;
```
[the remark this answers is not in the document]
Comment (elguardian) [reply kie-server-parent/kie-server-services/kie-server-services-jbpm-ui/src/main/java/org/kie/server/services/jbpm/ui/form/render/AbstractFormRenderer.java]: done

File: kie-server-parent/kie-server-services/kie-server-services-jbpm-ui/src/main/java/org/kie/server/services/jbpm/ui/form/render/AbstractFormRenderer.java [hunk 3]
```diff
@@ -407,24 +412,40 @@ protected void processFormLayout(FormInstance topLevelForm,
...
+                                                               .map(e -> (DocumentImpl) e)
+                                                               .forEach(DocumentImpl::load);
+
+                                    List<DocumentItem> items = docCollection.getDocuments()
+                                                                   .stream()
+                                                                   .map(e -> new DocumentItem(e.getName(), Base64.getEncoder().encodeToString(e.getContent())))
+                                                                   .collect(toList());
+                                    item.setValue(items);
+                                case "multipleSelector":
+                                    item.setOptions(field.getListOfValues().stream().map(e -> new ItemOption(e)).collect(toList()));
```
[the remark this answers is not in the document]
Comment (elguardian) [reply kie-server-parent/kie-server-services/kie-server-services-jbpm-ui/src/main/java/org/kie/server/services/jbpm/ui/form/render/AbstractFormRenderer.java]: done

File: kie-server-parent/kie-server-services/kie-server-services-jbpm-ui/src/main/java/org/kie/server/services/jbpm/ui/form/render/AbstractFormRenderer.java [hunk 4]
```diff
@@ -655,17 +676,14 @@ protected String getFieldType(String type) {
             return "Boolean(";
         } else if (type.contains("Date")) {
             return "Object(";
-        } else if (type.contains("Integer") || type.contains("Double") || type.contains("Float")) {
-            return "Number(";
-        } else if (type.contains("Boolean")) {
-            return "Boolean(";
-        } else if (type.contains("Document") || type.contains("documentCollection") || type.contains("Date")) {
+        } else if (type.contains("Document") || type.contains("documentCollection") || type.contains("multipleSelector") || type.contains("multipleInput") || type.contains("Date")) {
```
[the remark this answers is not in the document]
Comment (elguardian) [reply kie-server-parent/kie-server-services/kie-server-services-jbpm-ui/src/main/java/org/kie/server/services/jbpm/ui/form/render/AbstractFormRenderer.java]: done

File: kie-server-parent/kie-server-services/kie-server-services-jbpm-ui/src/main/resources/form-templates-providers/patternfly/input-form-group-template.html [hunk 5]
```diff
@@ -136,6 +136,29 @@ <h3 data-i18n-key="dropFilesHere">Drop files here</h3>
...
+    </select>
+</div>
+<#elseif item.type == "multipleInput">
+<div class="form-group">
+  <label for="${item.id}">${item.label}</label>
+  <#if item.required>
+    <strong style="color: red">*</strong>
+  </#if>
+  <input type="text" multiple class="form-control" id="${item.id}" ${(item.readOnly)?string('disabled', '')} data-role="tagsinput" value="${item.value}" />
+</div>
```
[the remark this answers is not in the document]
Comment (elguardian) [reply kie-server-parent/kie-server-services/kie-server-services-jbpm-ui/src/main/resources/form-templates-providers/patternfly/input-form-group-template.html:126]: done

**Review by afalhambra — APPROVED** [review]
Thanks @elguardian - nice job.

**Pull request merged**